src/libstd/sys/unix/ext/net.rs

```diff
@@ -680,7 +615,10 @@ impl SocketAncillary {
     /// ```
     #[unstable(feature = "unix_socket_ancillary_data", issue = "none")]
     pub fn clear(&mut self) {
-        self.buffer.clear();
+        unsafe {
+            libc::memset(self.buffer.as_mut_ptr().cast(), 0, self.buffer.len());
+        }
```

Comment:
This seems unnecessary since lengths are carried explicitly

Reply:
I have to zero the buffer. If data is appended in the function `add_to_ancillary_data` then the `self.length`value is increased and the last `cmsg` is determined to add the new data. The last `cmsg` is determined by calling the function `CMSG_NXTHDR` until it returns `null`. It has to be prevented that `CMSG_NXTHDR` interpret old `cmsg` data as current `cmsg`.

Reply:
Ah, this wasn't obvious to me from the linux cmsg documentation, but you're right, there was [some discussion ](https://www.austingroupbugs.net/view.php?id=1056) in the posix standards. Perhaps add a comment here?